### ISSUE TITLE
Gradual Infill Steps

### ISSUE THREAD

**Issue by TheAtomicSoul** [SoftFever/OrcaSlicer#7454]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Gradual Infill Steps: Both to infill and support!
This is the ONLY FEATURE I MISS from Cura: It both helps REDUCE WEIGHT and REDUCE MATERIAL USED. 

Allows better support of top surfaces where needed with MUCH less material use overall.

![image](https://github.com/user-attachments/assets/8a1341e7-0937-41f6-9508-2950d03a27ea)
![image](https://github.com/user-attachments/assets/ab734549-ffd4-40e2-8b8c-2e98665a13dc)
![image](https://github.com/user-attachments/assets/bb64d25d-df9e-4b63-9682-c684bb5e2d86)

![image](https://github.com/user-attachments/assets/2ea0b952-fffa-4880-be19-0739705405d1)

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

Add Gradual Infill Step options from Cura to both infill and support
- Gradual Infill Steps
- Gradual Infill Step Height

### Additional context

Previously submitted and closed by a bot:
- by me: https://github.com/SoftFever/OrcaSlicer/issues/3100
- by someone else: https://github.com/SoftFever/OrcaSlicer/issues/5083

**Comment by TheAtomicSoul** [comment]
Bump to keep from getting stale.

**Comment by tlhintoq** [comment]
Came to log the same request and it was already here.

Dear gawd yes... Would love to have an 80% infil that halves, halves, halves... So it starts at nearly nothing then becomes this nice density just below the external top surface layers.
I got nowhere with some of the infil patterns and was starting to plan staggered layers of modifiers like layers in a cake to achieve the same - but what a mess to set up especially on several things.

**Comment by Nandox7** [comment]
Creality added to it's later slier version (5.1) a feature called AI Infill that is someone what is asked here.
In summary more infill density near the walls ad less away from the walls.

https://www.youtube.com/watch?v=yUhf6j6wNY8

**Comment by tlhintoq** [comment]
That would be freaking awesome!

**Comment by TheAtomicSoul** [comment]
Bumping this to keep it alive. REALLY wish this would be implemented.

**Comment by coltography** [comment]
Was looking into this after discovering adaptive cubic and surprised its not a feature in orcaslicer already! Hope it can be added some day.

**Comment by hotellonely** [comment]
bumping again we REALLY need this instead of the poor adaptive cubic which is really poorly designed.

**Comment by tlhintoq** [comment]
**_Bandaid_**:  Use modifiers.

![Image](https://github.com/user-attachments/assets/7736983c-b829-47fd-bc91-44cbf53224c1)

If you want to get really crazy you can even change pattern with a modifier.
From 5% cubic to 15% hex if you want.

![Image](https://github.com/user-attachments/assets/0f262538-669b-461a-b9ab-f1dad925643e)

**Comment by Nandox7** [comment]
That's cool and will allow on a per layer change, but does not cover the need to have it change horizontally.
I imagine with even more work you can do several modifier blocks that get it to change the infill depending if it's closer to the all or more at the center, but it would be mad work and depends on how complex is the part.

**Comment by tlhintoq** [comment]
To be fair the Cura feature this would be cloning doesn't do a smooth transition either.
It doesn't change from 1% to 3% to 5%.
It does an increase of halves (doubling) for the entire part at the prescribed intervals.

**Comment by hotellonely** [comment]
> To be fair the Cura feature this would be cloning doesn't do a smooth transition either. It doesn't change from 1% to 3% to 5%. It does an increase of halves (doubling) for the entire part at the prescribed intervals.

Yes, but it's quite effective. In fact you don't need a 80% for perfect pillow free experience, in my real world tests usually 25% is already pretty much pillow free.

**Comment by ariefadha** [comment]
I really need this feature, hope someone will add this, thanks.

**Comment by TheAtomicSoul** [comment]
Adding another bump. I don't get how people don't see how useful this would be.

**Comment by TheAtomicSoul** [comment]
Monthly Bumpo.
I do think Creality slicer(5.1) feature called AI Infill mentioned above may be the best option.

**Comment by TheAtomicSoul** [comment]
Ultimaker Cura Gradual Infill Steps https://youtu.be/z038SdtIk0c I used this ALL THE TIME in Cura and it's my BIGGEST MISSED FEATURE.

Creality slicer(5.1)  AI Infill: https://youtu.be/yUhf6j6wNY8 (not really a good visual example, but surprisingly there isn't one?)

**Comment by ariefadha** [comment]
creality print has major update. and it has gradual 1st layer. top surface path and speed dependent on height.

https://wiki.creality.com/en/software/6-0/release-notes-6-3-0

i think i will move back to creality print for now

**Comment by TheAtomicSoul** [comment]
Surprised this hasn't been closed. Also surprised I haven't heard of it being implemented yet.

**Comment by inspectionsbybob** [comment]
Bump to keep this active.
This is the only reason I have 4-5 different slicers on my system.  Each one of them has 1-2 features that no one else has implemented, going all the way back to slic3D

There has been lots of work done on this so I am confused as to why no-one has imported it.

PLEASE, PLEASE, PLEASE
